pkg/registry/apis/dashboard/legacy_storage.go

```diff
@@ -83,6 +79,10 @@ func (s *storeWrapper) Create(ctx context.Context, obj runtime.Object, createVal
 		}
 	}
 
+	if err != nil {
```

Comment:
It's important to set the UID before returning to not let the wrong UID leak out in any way. 

So the err after the UID setting was intentional

Reply:
Updating. I don't understand why the obj is returned / consuming when there is an error 🤔 . I ll just trust you on that one.

Reply:
Me either, maybe some cleanup stuff? I didn't want to investigate that deep when I looked into it 😅